gusto/transport_equation.py

```diff
@@ -162,40 +228,45 @@ def __init__(self, state, V, *, ibp="once", equation_form="advective",
             raise ValueError("equation_form must be either 'advective' or 'continuity'")
         self.vector_manifold = vector_manifold
         self.outflow = outflow
-        if outflow and ibp is None:
+        if outflow and ibp == IntegrateByParts.NEVER:
             raise ValueError("outflow is True and ibp is None are incompatible options")
 
     def advection_term(self, q):
 
         if self.continuity:
-            if self.ibp == "once":
+            if self.ibp == IntegrateByParts.ONCE:
                 L = -inner(grad(self.test), outer(q, self.ubar))*dx
             else:
                 L = inner(self.test, div(outer(q, self.ubar)))*dx
         else:
-            if self.ibp == "once":
+            if self.ibp == IntegrateByParts.ONCE:
                 L = -inner(div(outer(self.test, self.ubar)), q)*dx
             else:
                 L = inner(outer(self.test, self.ubar), grad(q))*dx
 
-        if self.dS is not None and self.ibp is not None:
-            L += dot(jump(self.test), (self.un('+')*q('+')
-                                       - self.un('-')*q('-')))*self.dS
-            if self.ibp == "twice":
+        if self.dS is not None and self.ibp != IntegrateByParts.NEVER:
+            n = FacetNormal(self.state.mesh)
+            un = 0.5*(dot(self.ubar, n) + abs(dot(self.ubar, n)))
+
+            L += dot(jump(self.test), (un('+')*q('+')
+                                       - un('-')*q('-')))*self.dS
+
+            if self.ibp == IntegrateByParts.TWICE:
                 L -= (inner(self.test('+'),
-                            dot(self.ubar('+'), self.n('+'))*q('+'))
+                            dot(self.ubar('+'), n('+'))*q('+'))
                       + inner(self.test('-'),
-                              dot(self.ubar('-'), self.n('-'))*q('-')))*self.dS
+                              dot(self.ubar('-'), n('-'))*q('-')))*self.dS
 
         if self.outflow:
-            L += self.test*self.un*q*self.ds
+            n = FacetNormal(self.state.mesh)
+            un = 0.5*(dot(self.ubar, n) + abs(dot(self.ubar, n)))
+            L += self.test*un*q*(ds_v + ds_t + ds_b)
 
         if self.vector_manifold:
-            un = self.un
+            n = FacetNormal(self.state.mesh)
             w = self.test
-            u = q
-            n = self.n
             dS = self.dS
+            u = q
```

Comment:
Do we need this redefinition, or can we just use `inner(q('+'), n('+'))` etc below?

Reply:
I'm happy with `q` but @colinjcotter wrote this bit (although it looks like I've edited it here - not sure how/why!)

Reply:
I think it would only be used for velocity so `u` makes sense...